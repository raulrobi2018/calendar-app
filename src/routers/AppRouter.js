import React, {useEffect} from "react";
import {useDispatch} from "react-redux";

import {
    BrowserRouter as Router,
    Redirect,
    Switch,
    Route
} from "react-router-dom";

import {startChecking} from "../actions/auth";
import {LoginScreen} from "../components/auth/LoginScreen";
import {CalendarScreen} from "../components/calendar/CalendarScreen";

export const AppRouter = () => {
    const dispatch = useDispatch();

    //Solo tendrá como dependencia al dispatch porque el startChecking
    //no está dentro de este componente y por lo tanto no cambiará
    useEffect(() => {
        dispatch(startChecking());
    }, [dispatch]);

    return (
        <Router>
            <div>
                {/* A <Switch> looks through its children <Route>s and
         renders the first one that matches the current URL. */}
                <Switch>
                    {/* exact para que machee la ruta exacta */}
                    <Route exact path="/login" component={LoginScreen} />
                    <Route exact path="/" component={CalendarScreen} />

                    {/* Si no encuentra ningún path, lo redirige a calendar */}
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    );
};
