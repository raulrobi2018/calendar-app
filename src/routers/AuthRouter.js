import React from "react";
import {Redirect, Route, Switch} from "react-router-dom";
import {LoginScreen} from "../components/auth/LoginScreen";
import {RegisterScreen} from "../components/auth/RegisterScreen";

export const AuthRouter = () => {
    return (
        // Se utiliza doble guión bajo para indicar la carpeta donde se encuentra. Es una
        // convención de Sass
        <>
            <Switch>
                <Route exact path="/auth/login" component={LoginScreen} />
                <Route exact path="/auth/register" component={RegisterScreen} />

                {/* Si no encuentra ningún path, lo redirige al Login */}
                <Redirect to="/auth/login" />
            </Switch>
        </>
    );
};
