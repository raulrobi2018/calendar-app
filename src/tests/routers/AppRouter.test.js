import React from "react";
import {mount, configure} from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";

import {AppRouter} from "../../routers/AppRouter";

//Configuración necesaria para que funcione mount con React 17
configure({adapter: new Adapter()});
import configureStore from "redux-mock-store"; //ES6 modules
import thunk from "redux-thunk";

// Configuracion necesaria para probar dispatch
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

//Reemplaza la funcionalidad del dispatch del store por una función jest
// store.dispatch = jest.fn();

describe("Testing AppRouter component", () => {
    test("should display the Cargando...", () => {
        const initState = {
            auth: {
                checking: true
            }
        };
        const store = mockStore(initState);

        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <AppRouter />
                </MemoryRouter>
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find("h5").exists()).toBe(true);
    });
});
