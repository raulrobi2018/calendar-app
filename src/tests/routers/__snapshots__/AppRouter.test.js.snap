// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AppRouter component should display the Cargando... 1`] = `
<h1>
  Cargando
</h1>
`;
